### PR TITLE
Traitor uplink changes: Remove don't improve

### PR DESCRIPTION
**Pull request by Cocacolagua**
warning: wall of text

## What Does This PR Do

Removes the following items from the traitor uplink:
- False Bottomed Briefcase
- Clown Magboots
- Roboticist Syndicate MMI
- Poison Pen
- Power Beacon
- Single C4
- Bag of 10 C4s from nukies
- Chameleon Flag
- Adaptive Cardboard Cutouts

Reduces the cost of the following items:
- Chef Poison ( 2 -> 1) (considering just letting them buy poison bottles and removing this)
- Doctor Radiation Laser (5 -> 3)
- Virologist Viral Injector (3 -> 2)
- Dehydrated Carp (2 -> 1)
- Energized Fire-axe (10 -> 8)
- Nanocalcium Injector (4 -> 3) (considering reducing to 2)
- Radio Jammer (5 -> 2)
- 5 C4 kit (4 -> 3)
- Uplink Implant (14 -> 12)
- All Stechkin Ammo and .357 ammo(2 or 3 -> 1)

Makes the following items hijack-only:
- Librarian E20 (you can also no longer get it from surplus)
- Macrobomb

And whatever else:
- Allows traitors to buy X4 and microbomb implants, the latter being restricted to one for each uplink
- Replaces 3TC traitor drill with 1TC nukie diamond drill
- Incendiary ammo for the stechkin deals 25 damage instead of 30
- Hijack Syndicate bombs have a 30 second timer instead of a 90 seconds one (nukies still have the 90 seconds)

## Why It's Good For The Game

   All items I've removed I consider bloat to the uplink, given that they're unfixable in their current state. This is due to them either fulfilling a purpose that isn't that needed (clown magboots, poison pen, chameleon flag, cardboard cutouts, briefcase), are outclassed by something else (Syndicate MMI by the emag, C4 to X4), bear in mind you can still buy the kit of 5 C4s for specialized tactics with ninja stars and payloads, i just removed the excessive or barebones quantities or in the special case of the power beacon, it encourages hijackers to try the second most boring hijack tactic there is, next to e-magging bolted doors.

   The items i've reduced the price from have their uses, and can be salvaged from their shitty pricing, explanation for the reduction being:
- They have a good effect, but their price is just too high to be considered a good deal, this is the case with the nanocalcium injector, radio jammer, uplink implant, the different ammo for the stechkin and .357.
-  They're a subpar item with a bad effect for it's price, this is my way of seeing if they're actually worth anything before removing them or giving them a full rework, this is the case with chef sauce, radiation laser, energized fireaxe, dehydrated carp and viral injector.

Macrobomb was made hijack-only given that this will get you banned by using it outside of the hijack objective, same to the E20, given that rolling too high will get you in trouble. The timer was reduced from the macrobomb given that it's also a bad item as is, so now the chances of it being disarmed or thrown into space before explosion are much slimmer.

Microbomb implants were added as a tool for traitors to both get rid of soon-to-be-corpses or to get the final laugh on security. The implant has a limited stock of one per uplink, so you cannot stack it in the same way nuclear operatives can.

X4s are actually useful C4s, consider them a replacement of C4s (in the breaching aspect) after this PR , as they'll fulfill the purpose of breaching into places consistently without having to pop five of them.

I reduced the damage from incendiary ammo to give it a downside from normal ammo, to make it a side-grade like the other ammo types, whether it's dealing less damage to armored people or less damage in general, for the upside of armor penetration

## Changelog
:cl:
add: Added X4 and diamond vault drills to the traitor uplink.
add: Added Microbomb implants to the traitor uplink, they have a limited stock of one.
del: Removed clown magboots, syndicate MMI, viral injector, poison pen, power beacon, 1 C4 and 10 C4s, false bottomed briefcase, 3TC drill, chameleon flag and adaptive cardboard cutouts from the uplink.
tweak: Chef Poison costs 1TC instead of 2 in the uplink
tweak: Radiation laser costs 3TC instead of 5 in the uplink.
tweak: Dehydrated Carp cost 1TC instead of 2 in the uplink.
tweak: Energized Fire-axe costs 8TC instead of 10 in the uplink.
tweak: Nanocalcium injector costs 3TC instead of 4 in the uplink.
tweak: Radio Jammer costs 2TC instead of 5 in the uplink.
tweak: Uplink implant costs 12TC instead of 14 in the uplink.
tweak: All stechkin ammo and .357 ammo costs 1TC in the uplink.
tweak: the 5 C4 kit costs 3TC instead of 4 in the uplink.
tweak: Stechkin incendiary ammo deals 25 damage instead of 30.
tweak: Syndicate bomb and E20 are now hijack-only purchases from the uplink.
tweak: Syndicate bombs bought from the hijack uplink have a 30 second timer instead of a 90 seconds one.
/:cl:

### CODE REVIEW

**Comment by Cocacolagua** [comment]
whoops forgot to add this to the main text: to make incendiary ammo for the stechkin not a straight upgrade from the regular one, i reduced it's damage from 30 to 25.
there's also other changes not documented / not implemented, those will also be removed

**Comment by shmabuu** [comment]
I got a question.. when the game chooses what items to be discounted, does it chose from a list or does it look at what items are one TC and rules them out? And if it chooses one TC items to rule them out, would it be possible to do the same to items that are two TC? Getting a two TC item at a 50% discount just feels.. redundant. If it chooses from a list, make sure that the stechkin ammo can't be discounted. As of now the fact that it can be discounted is already pointless as is, but if it's not removed from this "List" and is now one TC, I think you can see my issue.

**Comment by datlo** [comment]
Removing c4 prevents the use of homemade plasteel bomb because you need for chemical payload. I've also bought and used c4 multiple times, it's definitely not useless if you know what you're buying them for.

Traitors should absolutely not get microbomb implants. Fun fact, they stack. Restricting macrobomb does nothing when you can just buy a bunch of them, inject them all, and get the same effect as a macrobomb. Even if you changed that, I just dont think traitor need that as a tool. 

On the viro injector and clown magboots, those are job-restricted items. They arent really bloat considering they wont show up for most traitors on station.  Their uses are niche, but not useless, especially the viro injector since standard stealth injection methods (syringe gun/sleepy pen) do not transmit viruses.

As the guy who made clown magboots... I understand why you'd remove them but I still think they're a useful niche item for clowns that dont want to give up their honking shoes. Maybe lower the cost to 1 or 2 so theyre more interesting to buy as a clown only item?


Other changes listed on PR description seem fine to me.

**Comment by Cocacolagua** [comment]
i completely forgot to mention i left the kit of 5 C4 for the purpose of payloads and pairing it up with ninja stars, sorry for being stupid.

on the topic of clown shoes however, the problem with them is that magboots themselves are already not a needed item, this wasn't the case before-hand when we had fastmos, but now the ability they serve doesn't really justify any cost, given that there's no space wind.

**Comment by datlo** [comment]
Yeah that's true. I think no slips even have magboots effect now. 

Hmmm... No slips really should give you waddling and honking if you transform them into clown shoes, then yeah itd definitely be a nonissue. Out of scope for this PR I guess, but something to keep in mind for the future.

**Comment by Cocacolagua** [comment]
CQC no longer a part of this PR, ignore the above link to this PR because i deleted that repository, i'll just bring it up another day

**Comment by Carthusia** [comment]
Poison pen, is that the sleepy pen?

**Comment by Cocacolagua** [comment]
@Carthusia poison pen is a job_specific item for the librarian, hop, cargo tech and QM, which poisons paper and anyone that reads it without gloves

**Comment by AzuleUtama** [comment]
We have way too much on the uplink that never gets used (usually for good reason), so these are overall some good changes in my eyes. I did have a few pointers though:

- I don't think the Syndi bomb should be restricted to hijack only - else it'll never get used by anyone due to the 90 second countdown combined with ease of spacing or disarming the bomb by people who know how. It makes a very obvious beep while active that only gets louder the closer it is to going off - so anyone paying attention will be able to move away or at least call it out. If we're seeing issues where they're being bought too often by traitors then raising the price might be a better option, we already barred them from surplus crates and discounts to tackle that.

- There's a stealthy item called the False Bottomed Briefcase which could honestly just be removed as well. I ported it a while back thinking it would be a neat way to disguise firearms in the open - but I've never seen it used by anyone, even when found in a surplus crate.

- I would also suggest having a look at the safe drills and possibly making traitors able to buy the diamond tipped one like nukies while removing the standard thermal drill from the uplink altogether. You can break into a safe with the stethoscope much faster than with the drill and it's either completely free or costs 1TC to get it compared to 3TC for a drill that takes forever and makes a tonne of noise in the process. Having the better drill be an option at the same price while keeping the shittier one as a craftable item would balance the two options out better.

**Comment by Cocacolagua** [comment]
Changes done:
- Brought back the viral injector and reduced it's price to 2TC, as i've been told it's the best item to infect people with a deadly virus that transmits through blood to avoid mass murder.
- Brought back traitors being able to buy Microbomb Implants, but this time gave them the limited_stock var, which only allows you to buy one per traitor uplink, nuclear operatives can still buy unlimited.

will also remove the false bottomed briefcase and change the drill to the nuclear operatives one.

@AzuleUtama the issue is that traitors are straight up banned from using the syndicate bomb without having the hijack objective, so nerfing the price wouldn't solve the issue. I will however make a sub-type for the hijack purchase of the syndicate bomb (nukie one will remain unchanged) to detonate at 30 seconds instead of 90, which makes it harder to space / disarm.

**Comment by Carthusia** [comment]
I still don't really see the need to be removing these items, why not simply improve them to make them more useful?

**Comment by Spacemanspark** [comment]
Wait, why remove the syndicate MMI?

**Comment by Dashsora** [comment]
I have bought chameleon flags 100% for the fluff (and even used one once as part of a surplus !), could it be part of the chameleon kit ? Don't think it'd be game breaking in any way, just cool to have !

**Comment by Oyu-De** [comment]
Out of curiosity, reason for removing poison pen?

**Comment by PacifistDalek** [comment]
Adaptive cardboard cutouts are just fun times, especially for making people think it's a non-traitor tound. Maybe just lower their price to see if anyone will use them?

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -383,6 +365,8 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/dice/d20/e20
 	cost = 3
 	job = list("Librarian")
+	surplus = 0 //hijack only so no surplus
```
Comment (farie82) [inline code/datums/uplink_item.dm:368]:
```suggestion
	surplus = FALSE //hijack only so no surplus
```

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -213,13 +205,24 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
+//Janitor
+
+/datum/uplink_item/jobspecific/cautionsign
+	name = "Proximity Mine"
+	desc = "An Anti-Personnel proximity mine cleverly disguised as a wet floor caution sign that is triggered by running past it, activate it to start the 15 second timer and activate again to disarm."
+	reference = "PM"
+	item = /obj/item/caution/proximity_sign
+	cost = 4
+	job = list("Janitor")
+	surplus = 0
```
Comment (farie82) [inline code/datums/uplink_item.dm:217]:
```suggestion
	surplus = FALSE
```

**Review by farie82 — CHANGES_REQUESTED** [review]
few minor nitpicks

**Comment by Rurik123** [comment]
power beacon still has a purpose does it not, even if hijack only? struggling to see reason for axing that one

**Comment by hal9000PR** [comment]
> Out of curiosity, reason for removing poison pen?

the poison pen (not the sleepy pen) is never used. It injects a tiny amount of toxin, and your target has to read paperwork, without wearing gloves, in order for it to work. basically, it is never used because it is useless.

**Comment by Spacemanspark** [comment]
I'm going to say that removing many of these things is a bit extreme. Even if some may be viewed as pointless for gaming your objectives, they can still be quite fun to use and/or see on station.

**Comment by Cocacolagua** [comment]
Alright, here's what I'm thinking:
- Allow poison pen to go through gloves except latex and nitrile.
- Leave the MMI alone for now.

And in the fence abiut:
- Add chameleon flag to the chameleon kit (the kit already has a ton of items, im skeptical of adding another one)

**Comment by Cocacolagua** [comment]
Wait wrong button mobile fuck

**Comment by hal9000PR** [comment]
> Wait wrong button mobile fuck

ahhh yes, gamer move.

**Comment by Pumpkin312** [comment]
false bottomed briefcase could be used for roleplay purposes i think it could stay.
syndie mmi would be much better if it was not roboticist exclusive (example, you find a mmi from a destroyed borg on the floor and you just put the brain inside the syndie mmi, or you break into roboticist's office early on and replace all mmis with syndie mmis.)
macrobomb i think should stay as is but i cant give you a good reason. maybe because its a classic.

**Comment by Rurik123** [comment]
> Reason I don’t believe the power beacon is a good addition to the uplink is because it encourages a hijack strategy that always ends in the same way:
> 
> * End the round early by releasing the engine.
> * Not know how that the beacon has to be on a wire to work, so it doesn’t even make it’s way to departures.
> * Or if the hijacker _did_ set up the power beacon correctly, they still fluke the actual hijack because they spent half of their budget on calling the shuttle, not arming themselves up against the crew.

Fair points. I'll miss the few times I've seen the powerbeacon set up to saw the station in half with a singularity, but alas it was rare and most of the time there was just botched engine releases. Discouraging engine releases since they are the laziest hijack option is a good thing in my book.

I agree with all the TC price reductions. Glad to see nanocalcium gradually be adjusted down over the year to a price that better reflects its usefulness.

**Comment by Oyu-De** [comment]
> > Out of curiosity, reason for removing poison pen?
> 
> 
> 
> the poison pen (not the sleepy pen) is never used. It injects a tiny amount of toxin, and your target has to read paperwork, without wearing gloves, in order for it to work. basically, it is never used because it is useless.

Now that you mention it ye, it is "poison pen" not "sleepy pen" my bad. Guess this also shows how rarely it is used

**Comment by Spacemanspark** [comment]
I assume by leaving the MMI alone, you're referring to it being left in the uplink?

**Comment by Cocacolagua** [comment]
@Spacemanspark pretty much yes

**Comment by Spacemanspark** [comment]
Alrighty.

**Comment by PacifistDalek** [comment]
I'd be curious to see if poison pen would be used more if it was buffed, with stronger poison and the ability to ignore most gloves.

**Comment by Carthusia** [comment]
Ooohh thatd be cool, basicly anything thats not rubber (Insulated gloves/Full blue latex gloves) allows the poision to seep in.

Another option would be just to merge the Poision pen and Sleepy pen together, allow it to both jab people and poision the paper

**Review by Kyep — APPROVED** [review]
While I am on the fence about some of this (e.g: adding X4, adding microbomb, increasing the damage of incin ammo, reducing cost of energized fireaxe, etc) there's quite a few things in here which are good (removing never-used uplink items, reducing cost of jammers and uplink implants, making bomb/E20 hijack-only, reducing timer on syndibomb). And there's nothing in here that I'd consider absolutely bad.
So overall, I'm leaning towards a yes on this.
If I had to make a suggestion, it would be:
- don't include microbombs in the uplink, even at a limited stock
- keep the syndi MMI
- don't buff incin ammo (or at least explain why you're buffing it)
- don't remove the sprites for things like power beacons (we may want to re-use those!)
- MAYBE don't reduce the price of carp all the way down to 1 (what if someone buys 20 of them?)

**Comment by Cocacolagua** [comment]
@Kyep some comments:
- incinerator ammo's price was reduced from 2 to 1, but since it was a straight upgrade from normal ammo, I lowered it’s damage from 30 to 25.
- I reduced carp's price to 1 since it's very ineffective at 2, with everything else being a better tool to slowdown security. Although I could make a bundle of 5 carps at the cost of 6, which means at most you’d get 15 carps.

**Comment by AzuleUtama** [comment]
One thing I've remembered regarding the "limited_stock" variable is that it only limits the number of purchases for an uplink device, not the player themselves. If they get access to more uplinks (uplink implant, stealing or acquiring multiple pdas and using the F.R.A.M.E cartridge on them, asking other traitors to buy you one etc.) then they can bypass this restriction. The same does apply for discounts since it was something I brought up when I ported them over ages ago - but it wasn't deemed an issue since they're random for each uplink whereas a microbomb implant would always be purchasable with each new uplink you get.

This meant if someone got 5 spare pdas, Frame'd them all and bought microbomb implants on each plus the one on their initial device they'd have 6 in them pretty easily, since the cartridge only costs 4TC on its own.

**Review by variableundefined — CHANGES_REQUESTED** [review]
Address Farie concerns

**Comment by variableundefined** [comment]
Also deconflict I guess

**Comment by Cocacolagua** [comment]
aight

**Comment by Muhdi345** [comment]
I don't get the hate for the power beacon. Engine sabotage generally happens because you want to force a shuttle, and how you deal with the aftermath is up to you.

**Comment by hal9000PR** [comment]
Engine sabotage hijack sucks. It is last and it doesn't help the hijacker outside of getting the shuttle called, which can also be attained by waiting. So it fucks over all other traitors and people who have big plans for that shift, all because you cannot wait. It's a poor hijack strategy and it shouldn't be encouraged.

**Comment by Spacemanspark** [comment]
On the other hand, the occasional engine release can be very interesting.
I wish there was a compromise, but alas.

**Comment by Pixelzilla** [comment]
I dont really get why some of these things need to be removed just because some people dont use them doesn't mean no one does and it doesn't mean they cant be useful (or hilarious) when used in a creative way just because there are "better" options or they dont suit a super serious playstyle.

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -995,19 +942,29 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
-	desc = "The Syndicate Bomb has an adjustable timer with a minimum setting of 90 seconds. Ordering the bomb sends you a small beacon, which will teleport the explosive to your location when you activate it. \
+	desc = "The Syndicate Bomb has an adjustable timer with a minimum setting of 30 seconds. Ordering the bomb sends you a small beacon, which will teleport the explosive to your location when you activate it. \
 	You can wrench the bomb down to prevent removal. The crew may attempt to defuse the bomb."
 	reference = "SB"
-	item = /obj/item/radio/beacon/syndicate/bomb
+	item = /obj/item/radio/beacon/syndicate/bomb/hijack
 	cost = 11
 	surplus = 0
 	cant_discount = TRUE
+	hijack_only = TRUE
```
Comment (Fox-McCloud) [inline code/datums/uplink_item.dm:957]: Really disagree with this one.

**Comment by Cocacolagua** [comment]
I don’t have time to support this PR as it is, given that it’ll have to defend my position for months, maybe I’ll atomize it sometime in the future

**Pull request closed without merging**